### PR TITLE
Python env

### PR DESCRIPTION
**Pull request by ArtoLord**

### CODE REVIEW

File: pylzy/lzy/py_env/py_env_provider.py [hunk 1]
```diff
@@ -87,6 +87,9 @@ def search(obj: Any) -> None:
             if name in STDLIB_LIST or name in sys.builtin_module_names:
                 return
 
+            if name in exclude_packages:
```
Comment (vhaldemar) [inline pylzy/lzy/py_env/py_env_provider.py:90]: I'd recommend to cast exclude_packages to set at some place for explicit O(1) in

File: lzy/worker/src/main/java/ai/lzy/worker/env/AuxEnvironment.java [hunk 2]
```diff
@@ -1,10 +1,65 @@
...
+            Files.createDirectories(localModulesPath);
+        } catch (IOException e) {
+            String errorMessage = "Failed to create directory to download local modules into;\n"
+                + "  Directory name: " + localModulesPath + "\n";
+            log.error(errorMessage);
+            throw new EnvironmentInstallationException(errorMessage);
+        }
+        var localModulesAbsolutePath = localModulesPath.toAbsolutePath();
+
+        log.info("CondaEnvironment::installPyenv created directory to download local modules into");
```
Comment (imakunin) [inline lzy/worker/src/main/java/ai/lzy/worker/env/AuxEnvironment.java:38]: Conda or Aux?

Comment (imakunin) [reply lzy/worker/src/main/java/ai/lzy/worker/env/AuxEnvironment.java:38]: in any case, this log message is incorrect :)

File: lzy/worker/src/main/java/ai/lzy/worker/env/SimplePythonEnvironment.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SimplePythonEnvironment implements AuxEnvironment {
```
Comment (imakunin) [inline lzy/worker/src/main/java/ai/lzy/worker/env/SimplePythonEnvironment.java:16]: let's avoid such a general words as `simple`, `common`, `complex`, ... :)

here you may wish to use `PurePythonEnv`, for example

Comment (imakunin) [reply lzy/worker/src/main/java/ai/lzy/worker/env/SimplePythonEnvironment.java:16]: or `PlainPythonEnv` as you use in logs below

File: pylzy/tests/api/v1/test_op_params.py [hunk 4]
```diff
@@ -279,47 +279,6 @@ def foo() -> None:
         call: LzyCall = wf.owner.runtime.calls[0]
         self.assertEqual(call.env.env_variables, {"a": "a1", "b": "b", "c": "c"})
 
-    def test_docker_only(self):
```
Comment (imakunin) [inline pylzy/tests/api/v1/test_op_params.py:282]: hmm... we need tests for the new functionality :)

File: lzy/worker/src/main/java/ai/lzy/worker/StreamQueue.java [hunk 5]
```diff
@@ -288,4 +283,63 @@ public void close() {
...
+        void logErr(String pattern, Object... values);
+
+        CompletableFuture<Void> logErr(InputStream stream);
+
+        void addErrOutput(OutputStream stream);
+
+        void addOutOutput(OutputStream stream);
+
+        static LogHandle fromTopicDesc(Logger logger, String taskId, @Nullable KafkaTopicDescription topicDesc,
+                                              @Nullable KafkaHelper helper)
```
Comment (imakunin) [inline lzy/worker/src/main/java/ai/lzy/worker/StreamQueue.java:301]: wrong indent

File: lzy/worker/src/main/java/ai/lzy/worker/env/AuxEnvironment.java [hunk 6]
```diff
@@ -1,10 +1,65 @@
...
+
+            extractFiles(tempFile, localModulesAbsolutePath.toString(), log);
+            tempFile.deleteOnExit();
+        }
+
+        return localModulesAbsolutePath;
+    }
+
+    private static void extractFiles(File file, String destinationDirectory, Logger log) throws IOException {
+        log.debug("Trying to unzip module archive "
```
Comment (imakunin) [inline lzy/worker/src/main/java/ai/lzy/worker/env/AuxEnvironment.java:59]: replcae `+` with log formatting, plz

**Review by imakunin — APPROVED** [review]

**Pull request merged**